### PR TITLE
refactor: [M3-6350] - MUI v5 Migration - `Components > InlineMenuAction`

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Refactors `InlineMenuAction` by removing jss and cleaning up the overall usage of the component.

## Major Changes 🔄
- `Drawer.jsx` now using `IconButton` instead of `Button` for the close component
- Removed excessive styles for `InlineMenuAction` component
   - Created a `StyledActionButton` until we can figure out what we want to do with this type of button. It doesn't follow our Button standards.
- Updated imports/exports
- `Button.tsx`: `compactX` and `compactY` should be something that's capable of being applied to any button type. In addition, we need UX to look at adding padding for when we use these props. At times the button text comes awfully close to the edge of the button which is not ideal.
   - I removed these props to anything that wasn't applied to `buttonType='secondary'` since it had no effect previously, it was essentially a dead prop.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-06-15 at 4 33 57 PM | No Change |
| Screenshot 2023-06-15 at 4 33 47 PM | No Change |
|  |  |

## How to test 🧪
1. Ensure there's no discrepencies for InlineMenuAction throughout site
2. Go to: http://localhost:3000/domains, http://localhost:3000/account/billing

### CODE REVIEW

**Comment by jdamore-linode** [comment]
> [`How did these get in here...`](https://github.com/linode/manager/pull/9268/commits/b2e9c0ac3d292da6ee6e893eb9bf344b686515d2)

That might've been my fault! These files have been around a while but I've definitely committed changes to them without thinking about it.

Can anyone think of a reason we couldn't add them to `.gitignore` so they don't get inadvertently re-added next time someone runs `yarn build`? (cc @bnussman-akamai)

**Comment by bnussman-akamai** [comment]
@jdamore-linode That makes sense to me in theory but we have some files that depend on the cached JSON. For example `packages/manager/src/__data__/ExtendedType.ts`. Until we refactor this stuff to use factories, I think we should probably keep it tracked by git.

**Review by bnussman-akamai — APPROVED** [review]
Looks good! Nice cleanup and fixes to the `Button`.

It would be really nice if we could add a basic default Storybook story for `<InlineMenuAction />` just so we have it documented.

File: packages/manager/src/components/InlineMenuAction/InlineMenuAction.tsx [hunk 1]
```diff
@@ -66,28 +27,29 @@ const InlineMenuAction: React.FC<CombinedProps> = (props) => {
...
   if (href) {
     return (
-      <Link className={`${className} ${classes.linkRoot}`} to={href}>
+      <StyledLink className={className} to={href}>
         <span>{actionText}</span>
-      </Link>
+      </StyledLink>
     );
   } else {
     return (
```
[the remark this answers is not in the document]
Comment (jaalah-akamai) [reply packages/manager/src/components/InlineMenuAction/InlineMenuAction.tsx]: Good call

**Comment by jaalah-akamai** [comment]
Screenshot 2023-06-29 at 2 52 39 PM

File: packages/manager/src/components/InlineMenuAction/InlineMenuAction.tsx [hunk 2]
```diff
@@ -1,60 +1,29 @@
...
-      },
-    },
-  },
-  linkRoot: {
-    textAlign: 'center',
-    color: theme.textColors.linkActiveLight,
-    borderRadius: 0,
-    '&:hover, &:focus': {
-      backgroundColor: '#3683dc',
-      color: '#ffffff',
```
Comment (mjac0bs) [inline packages/manager/src/components/InlineMenuAction/InlineMenuAction.tsx:38]: Noticed a regression on this branch where the button text color is not white on hover, which is evident in dark mode. Caused by removing some necessary styling?

This Branch:

https://github.com/linode/manager/assets/114685994/fb963938-8947-4f27-b627-dc6fcaf0e306




Prod:

https://github.com/linode/manager/assets/114685994/2150ecbd-562f-40db-992f-d9ca49944446

File: packages/manager/src/components/Button/StyledActionButton.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  label: 'StyledActionButton',
+})(({ theme }) => ({
+  fontFamily: latoWeb.normal,
+  fontSize: '14px',
+  lineHeight: '16px',
+  minWidth: 0,
+  padding: '12px 10px',
+  '&:hover': {
+    backgroundColor: theme.palette.primary.main,
+    color: theme.color.white,
```
Comment (mjac0bs) [inline packages/manager/src/components/Button/StyledActionButton.ts:21]:
```
    color: theme.color.white,
```
The culprit for this?

Screenshot 2023-06-30 at 1 26 33 PM

**Review by mjac0bs — COMMENTED** [review]
Thanks for the story and clean up! Noticed a couple of small regressions in dark mode. 🕶️

**Comment by jaalah-akamai** [comment]
@mjac0bs This should be good now. Also fixed the secondary button color... the hover/pressed need UX input eventually.

| Before      | After |
| ----------- | ----------- |
|Screenshot 2023-07-03 at 10 32 44 AM|Screenshot 2023-07-03 at 10 32 38 AM|

After:
Screenshot 2023-07-03 at 10 38 21 AM

**Review by mjac0bs — APPROVED** [review]

File: packages/manager/src/components/Button/StyledActionButton.ts [hunk 4]
```diff
@@ -18,6 +18,6 @@ export const StyledActionButton = styled(Button, {
   padding: '12px 10px',
   '&:hover': {
     backgroundColor: theme.palette.primary.main,
-    color: theme.color.white,
+    color: theme.name === 'dark' ? theme.color.black : theme.color.white,
```
[the remark this answers is not in the document]
Comment (jaalah-akamai) [reply packages/manager/src/components/Button/StyledActionButton.ts:21]: Yea... it's not great. We need to sit with Design and resolve this eventually

**Pull request merged**